starter-code/scripts/models/article.js

```diff
@@ -49,16 +49,31 @@ Article.fetchAll = function() {
     /* When our data is already in localStorage:
     1. We can process and load it,
     2. Then we can render the index page.  */
+    var ipsum = JSON.parse(localStorage.getItem('hackerIpsum'));
+    // for (var i of ipsum) {
+    //   new Article(i.keys);
+    //   console.log('the if has executed');
+    // }
+    articleView.renderIndexPage();
   } else {
     /* Without our localStorage in memory, we need to:
     1. Retrieve our JSON file with $.getJSON
       1.a Load our json data
       1.b Store that data in localStorage so that we can skip the server call next time,
       1.c And then render the index page.*/
-  }
-};
-
-
+      $.getJSON('../../data/hackerIpsum.json1').then(
+      function(data) {
+        console.log(data);
+      },
+      function() {
+        //console.error('This did not work');
+        console.log('This did not work');
+        }
+      );
+      articleView.renderIndexPage();
+      console.log('the else has executed');
```

Comment:
Currently, your getJSON call is not working. You can structure this multiple ways to get it to work, but essentially, all you're done once the data has been fetched is console.logging it. Any time we do an AJAX call (including $.getJSON), this is an asynchronous operation. If we want to do anything with the data coming back from the AJAX call, we have to do it once we are successful in receiving it. If we try to process the data outside of the "success" callback function passed to $.getJSON, it will not be available. Functions that deal with making requests to an outside resource, whether it is a file in the file system or a server or a database take time. The rest of our code will execute in the time it takes to receive a response. Therefore, if you try to access the data outside of a callback function (that is only invoked upon success), it will not yet be available.